File: bjtu_oj/CPP/c352B.cpp
#include<bits/stdc++.h>
using namespace std;

typedef long long ll;

const int maxn=2e6+5;

int a[maxn];

int dp[maxn];

int main() {
	int n,x,y;
	cin>>n>>x>>y;
	
	for(int i=1; i<=n; i++) {
		scanf("%d",&a[i]);
		ll r=(1ll<<61);
		ll l=0;
		while(l<r){
		    ll mid=(l+r)/2;
		    if(mid/x+mid/y>=a[i]){
		        r=mid;
            }else {
                l=mid+1;
            }
        }
        if(r%x==0&&(r%y==0)){
            printf("Both\n");
        }else if(r%x==0){
            printf("Vova\n");
        }else {
            printf("Vanya\n");
        }
	}

	return 0;
}

File: bjtu_oj/CPP/c352B.go
package main

import "fmt"

func main() {
	var n, x, y int64
	fmt.Scan(&n, &x, &y)
	for i := 1; int64(i) <= n; i++ {

		var l int64 = 0
		var r int64 = (1 << 60)
		var ai int64
		fmt.Scan(&ai)
		for l < r {
			var mid = (l + r) / 2
			if mid/x+mid/y >= ai {
				r = mid
			} else {
				l = mid + 1
			}
		}
		if r%x == 0 && (r%y == 0) {
			fmt.Printf("Both\n")
		} else if r%x == 0 {
			fmt.Printf("Vova\n")
		} else {
			fmt.Printf("Vanya\n")
		}
	}
}
